scripts/launch.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+export HOSTNAME_IP_ADDRESS=$(hostname -i)
+
+EXT=${QDROUTERD_CONF_TYPE:-conf}
+CONFIG_FILE=/tmp/qdrouterd.${EXT}
+
+${QDROUTERD_HOME}/bin/configure.sh ${QDROUTERD_HOME} $CONFIG_FILE
+
+if [ -f $CONFIG_FILE ]; then
+    ARGS="-c $CONFIG_FILE"
+fi
+
+if [[ $QDROUTERD_DEBUG = "gdb" ]]; then
+    exec gdb -batch -ex "run" -ex "bt" --args qdrouterd $ARGS
+elif [[ $QDROUTERD_DEBUG = "valgrind" ]]; then
+    exec valgrind qdrouterd $ARGS
+else
```

Comment:
Ha, here's the `--router-debug-mode`.

Few thoughts:

* wouldn't `catchsegv` do the backtrace printing just as well as gdb, only with less dependencies?
* or possibly `SEGFAULT_SIGNALS="all" LD_PRELOAD=/lib64/libSegFault.so`
* it would be probably also beneficial to compile with `-fstack-protector-all` and `-D_FORTIFY_SOURCE=2`, I sometimes saw glibc print stacktrace on its own
* alternatively, what about compiling with `-fsanitize=address`? That adds runtime checks and also a backtrace printing handler.

Reply:
* `LIBC_FATAL_STDERR_=1`

Reply:
Is there any overhead for -fsanitize=address?

Using catchsegv might well be fine (though I think we might want to include gdb anyway). I'd suggest making those changes as separate PRs though, and focus initially on removing the need for the interconnectedcloud image in this one.

Reply:
> Is there any overhead for -fsanitize=address?

Yes, something in the order of 2x or 4x for both memory and CPU. Plus at the start there is huge virtual memory allocation on the order of 20TB. Only a fraction of memory is not actually in use, but it means that if there are memory limits (ulimit, or k8s limits), they will probably be hit by this.

>  I'd suggest making those changes as separate PRs though

I never said otherwise. It was just that here in the PR all the image code is plain to see and easy to comment on, so that's what I was doing.

Reply:
-fstack-protector-all and -D_FORTIFY_SOURCE=2 have much less overhead; and afaik these are default cflags on ubuntu, and definitely when building rpm packages

Reply:
follow up issue https://github.com/skupperproject/skupper-router/issues/60